### PR TITLE
Fix MAC specific test failure.

### PR DESCRIPTION
**Pull request by wggihan**
## Purpose
Client socket related unit tests are failing on MAC. This is working fine on an Ubuntu and Travis CI.

### CODE REVIEW

File: tests/ballerina-test/src/test/java/org/ballerinalang/test/nativeimpl/functions/io/socket/ClientSocketTest.java [hunk 1]
```diff
@@ -29,34 +29,68 @@
...
+        try {
+            server = MockSocketServer.start();
+        } catch (IOException | InterruptedException e) {
+            log.error("Unable to open Socket Server: " + e.getMessage(), e);
+            Assert.fail(e.getMessage());
+        }
+        int retryCounter = 0;
+        int threshold = 10;
+        Socket tempSocket = null;
+        while (retryCounter < threshold) {
```
Comment (pamod) [inline tests/ballerina-test/src/test/java/org/ballerinalang/test/nativeimpl/functions/io/socket/ClientSocketTest.java]: @wggihan we can improve this by introducing a isConnected method ?

File: tests/ballerina-test/src/test/java/org/ballerinalang/test/nativeimpl/functions/io/socket/MockSocketServer.java [hunk 2]
```diff
@@ -37,76 +39,59 @@
...
-        if (serverSocket != null) {
-            try {
-                serverSocket.close();
-            } catch (IOException e) {
-                //Do nothing.
-            }
+    private static void answerWithEcho(ByteBuffer buffer, SelectionKey key) throws IOException {
+        SocketChannel client = (SocketChannel) key.channel();
+        client.read(buffer);
+        if (new String(buffer.array()).trim().equals(POISON_PILL)) {
```
Comment (pamod) [inline tests/ballerina-test/src/test/java/org/ballerinalang/test/nativeimpl/functions/io/socket/MockSocketServer.java]: Can we flip this statement ?

**Comment by pamod** [comment]
@wggihan let's hold onto this PR until we incorporate the changes. Temporarily disabled the test cases #4957 and created an issue #4956 for tracking

**Comment by wggihan** [comment]
This fixed through https://github.com/ballerina-lang/ballerina/pull/4970

**Pull request closed without merging**